curvefs/test/mds/mds_service_test.cpp

```diff
@@ -151,332 +314,198 @@ class MdsServiceTest : public ::testing::Test {
     std::shared_ptr<MockSpaceManager> spaceManager_;
     std::shared_ptr<MetaserverClient> metaserverClient_;
     std::shared_ptr<MockKVStorageClient> kvstorage_;
+    std::shared_ptr<MockVolumeSpace> volumeSpace;
     std::shared_ptr<MockTopologyManager> topoManager_;
     std::shared_ptr<MockS3Adapter> s3Adapter_;
-};
-
-template <typename RpcRequestType, typename RpcResponseType,
-          bool RpcFailed = false>
-void RpcService(google::protobuf::RpcController* cntl_base,
-                const RpcRequestType* request, RpcResponseType* response,
-                google::protobuf::Closure* done) {
-    if (RpcFailed) {
-        brpc::Controller* cntl = static_cast<brpc::Controller*>(cntl_base);
-        cntl->SetFailed(112, "Not connected to");
-    }
-    done->Run();
-}
-
-TEST_F(MdsServiceTest, test1) {
-    brpc::Server server;
-    // add metaserver service
-    MdsServiceImpl mdsService(fsManager_, nullptr);
-    ASSERT_EQ(server.AddService(&mdsService, brpc::SERVER_DOESNT_OWN_SERVICE),
-              0);
-
-    FakeMetaserverImpl metaserverService;
-    ASSERT_EQ(
-        server.AddService(&metaserverService, brpc::SERVER_DOESNT_OWN_SERVICE),
-        0);
+    std::shared_ptr<MdsService_Stub> stub_;
+    std::shared_ptr<MdsServiceImpl> mdsService_;
 
     MockCliService2 mockCliService2;
-    ASSERT_EQ(
-        server.AddService(&mockCliService2, brpc::SERVER_DOESNT_OWN_SERVICE),
-        0);
-
-    FakeCurveFSService fakeCurveFSService;
-    ASSERT_EQ(0, server.AddService(&fakeCurveFSService,
-                                   brpc::SERVER_DOESNT_OWN_SERVICE));
-
-    // start rpc server
-    brpc::ServerOptions option;
-    std::string addr = "127.0.0.1:6703";
-    std::string leader = "127.0.0.1:6703:0";
-    ASSERT_EQ(server.Start(addr.c_str(), &option), 0);
-
-    // init client
     brpc::Channel channel;
-    ASSERT_EQ(channel.Init(server.listen_address(), nullptr), 0);
-
-    MdsService_Stub stub(&channel);
+    brpc::ServerOptions option;
+    brpc::Server server;
     brpc::Controller cntl;
+    GetLeaderResponse2 getLeaderResponse;
+    FsInfo fsinfo1;
+    Volume volume;
+    FsInfo fsinfo2;
+    S3Info s3info;
 
-    // test CreateFS
-    CreateFsRequest createRequest;
-    CreateFsResponse createResponse;
+    uint64_t FS_CAPACITY = 100ULL << 30;
+    std::string S3_FS_NAME = "fs2";
+    std::string HOST_1 = "host1";
+    std::string HOST_2 = "host2";
+    std::string PATH_ROOT_A_B_C = "/a/b/c";
+    std::string PATH_ROOT_A_B_D = "/a/b/d";
+    std::string FS_OWNER = "test";
+    std::string VOLUME_FS_NAME = "fs1";
+    std::string CLUSTER_ADDR = "127.0.0.1:6703";
+    std::string RPC_SERVER_ADDR = "127.0.0.1:6703";
+    std::string META_SERVER_ADDR = "127.0.0.1:6703";
+    std::string RPC_SERVER_LEADER_ADDR = "127.0.0.1:6703:0";
+    std::set<std::string> COPYSET_ADDRS;
+};
 
+TEST_F(MdsServiceTest, test_fail_create_fs_missing_volume_for_volume_fstype) {
+    cntl.Reset();
```

Comment:
you can put `Reset` at the end of `SetUp`

Reply:
done